File: PythonMade/Ex 1/ex1.py
import numpy as np
import matplotlib.pyplot as plot

fund_f = 60                         # 60 Hz de frequência fundamental

sample_freqs = [480,960,1920,3840]  # Frequências de amostragem

amp = 1                             # Amplitude do sinal

n_cycles = 2                        # Número de períodos a serem amostrados

for fs in sample_freqs:
        
        # Tempo discretizado para cada amostra
        # Cria um vetor com cada ponto no tempo
        # Amostra a cada 1/fs segundos (inverso da frequência de amostragem)
        disc_t = np.arange(0,n_cycles * 1/fund_f, 1/fs)

        # Gera o sinal senoidal com as frequências desejadas
        signal = amp * np.sin(2 * np.pi * fund_f * disc_t)

        # Plota o sinal
        plot.figure()
        plot.stem(disc_t,signal,use_line_collection=True)
        plot.title(f"Senoide Discreta 60 Hz - Freq. de Amostragem: {fs} Hz.")
        plot.xlabel("Tempos (s)")
        plot.ylabel("Amplitude")
        plot.grid(True)

        # Salva as imagens no drietório do script
        plot.savefig(f"Senoide Discreta 60 Hz - Freq. de Amostragem: {fs} Hz.png")

plot.show() # Exibe os gráficos na tela

File: Ex 1/MatlabMade/ex1.m
clear;
close all;
clc;

P = 2; %Períodos dos sinal
A = 1; %Amplitude do sinal
fund_freq = 60; %Frequência fundamental do sinal

%Vetor de frequências de amostragem
sample_freqs = [480 960 1920 3840];

%Calcula o tempo total de amostragem do sinal
total_time = 1/fund_freq * P;

%Percorre todas as frequências de amostragem definidas no vetor
for i = 1:length(sample_freqs)
    sf = sample_freqs(i);

    %Vetor que contém as timestamps
    %Formato vetor = inicio:passo:fim ou inicio:fim com passo 1
    %Neste caso, o vetor começa em zero e vai até o último timestamp, com
    %passo 1, logo, dividimos o passo por sf pra ter o timestamp discreto em
    %segundos
    disc_t = (0:(total_time * sf) - 1) / sf;

    %Sinal discretizado
    sinal = A * sin (2 * pi * fund_freq * disc_t);

    %Plota os gráficos um abaixo do outro, para cada frequência de
    %amostragem
    figure(i)
    plot(disc_t,sinal,'lineWidth',2);
    xlabel('Tempo');
    ylabel('Amplitude');
    title(sprintf('FS = %d Hz',sf))
    grid on;
end